From 45603eeabc78e4eb4737110890bfe9f8e8f05180 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Mon, 11 May 2020 20:12:55 +0300
Subject: [PATCH 69/88] add speed UI overlays

	modified:   packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayManager.java
	modified:   packages/overlays/Android.mk
	new file:   packages/overlays/Level1SpeedUIOverlay/Android.mk
	new file:   packages/overlays/Level1SpeedUIOverlay/AndroidManifest.xml
	new file:   packages/overlays/Level1SpeedUIOverlay/res/values/config.xml
	new file:   packages/overlays/Level2SpeedUIOverlay/Android.mk
	new file:   packages/overlays/Level2SpeedUIOverlay/AndroidManifest.xml
	new file:   packages/overlays/Level2SpeedUIOverlay/res/values/config.xml
---
 .../systemui/theme/ThemeOverlayManager.java   |  8 ++++-
 packages/overlays/Android.mk                  |  2 ++
 .../overlays/Level1SpeedUIOverlay/Android.mk  | 30 ++++++++++++++++
 .../Level1SpeedUIOverlay/AndroidManifest.xml  | 24 +++++++++++++
 .../res/values/config.xml                     | 28 +++++++++++++++
 .../overlays/Level2SpeedUIOverlay/Android.mk  | 30 ++++++++++++++++
 .../Level2SpeedUIOverlay/AndroidManifest.xml  | 24 +++++++++++++
 .../res/values/config.xml                     | 34 +++++++++++++++++++
 8 files changed, 179 insertions(+), 1 deletion(-)
 create mode 100644 packages/overlays/Level1SpeedUIOverlay/Android.mk
 create mode 100644 packages/overlays/Level1SpeedUIOverlay/AndroidManifest.xml
 create mode 100644 packages/overlays/Level1SpeedUIOverlay/res/values/config.xml
 create mode 100644 packages/overlays/Level2SpeedUIOverlay/Android.mk
 create mode 100644 packages/overlays/Level2SpeedUIOverlay/AndroidManifest.xml
 create mode 100644 packages/overlays/Level2SpeedUIOverlay/res/values/config.xml

diff --git a/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayManager.java b/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayManager.java
index 339560139c1..fe6f22d3883 100644
--- a/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayManager.java
+++ b/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayManager.java
@@ -79,6 +79,9 @@ class ThemeOverlayManager {
     @VisibleForTesting
     static final String OVERLAY_CATEGORY_QSALPHA =
             "android.theme.customization.qsalpha";
+    @VisibleForTesting
+    static final String OVERLAY_CATEGORY_SPEEDUI =
+            "android.theme.customization.speedui";
 
     /*
      * All theme customization categories used by the system, in order that they should be applied,
@@ -93,6 +96,7 @@ class ThemeOverlayManager {
             OVERLAY_CATEGORY_STATUSBAR,
             OVERLAY_CATEGORY_QSALPHA,
             OVERLAY_CATEGORY_DARKUI,
+            OVERLAY_CATEGORY_SPEEDUI,
             OVERLAY_CATEGORY_ICON_ANDROID,
             OVERLAY_CATEGORY_ICON_SYSUI,
             OVERLAY_CATEGORY_ICON_SETTINGS,
@@ -108,6 +112,7 @@ class ThemeOverlayManager {
             OVERLAY_CATEGORY_STATUSBAR,
             OVERLAY_CATEGORY_QSALPHA,
             OVERLAY_CATEGORY_DARKUI,
+            OVERLAY_CATEGORY_SPEEDUI,
             OVERLAY_CATEGORY_ICON_ANDROID,
             OVERLAY_CATEGORY_ICON_SYSUI);
 
@@ -131,7 +136,7 @@ class ThemeOverlayManager {
                 OVERLAY_CATEGORY_SHAPE, OVERLAY_CATEGORY_ICON_ANDROID, OVERLAY_CATEGORY_DARKUI));
         mTargetPackageToCategories.put(SYSUI_PACKAGE,
                 Sets.newHashSet(OVERLAY_CATEGORY_ICON_SYSUI, OVERLAY_CATEGORY_QSPANEL,
-                                OVERLAY_CATEGORY_QSALPHA));
+                                OVERLAY_CATEGORY_QSALPHA, OVERLAY_CATEGORY_SPEEDUI));
         mTargetPackageToCategories.put(SETTINGS_PACKAGE,
                 Sets.newHashSet(OVERLAY_CATEGORY_ICON_SETTINGS));
         mTargetPackageToCategories.put(mLauncherPackage,
@@ -144,6 +149,7 @@ class ThemeOverlayManager {
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_ICON_ANDROID, ANDROID_PACKAGE);
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_STATUSBAR, ANDROID_PACKAGE);
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_DARKUI, ANDROID_PACKAGE);
+        mCategoryToTargetPackage.put(OVERLAY_CATEGORY_SPEEDUI, SYSUI_PACKAGE);
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_QSPANEL, SYSUI_PACKAGE);
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_QSALPHA, SYSUI_PACKAGE);
         mCategoryToTargetPackage.put(OVERLAY_CATEGORY_ICON_SYSUI, SYSUI_PACKAGE);
diff --git a/packages/overlays/Android.mk b/packages/overlays/Android.mk
index 3255ada35be..30feca5a783 100644
--- a/packages/overlays/Android.mk
+++ b/packages/overlays/Android.mk
@@ -81,6 +81,8 @@ LOCAL_REQUIRED_MODULES := \
 	StatusBar1Overlay \
 	StatusBar2Overlay \
 	StatusBar3Overlay \
+	Level1SpeedUIOverlay \
+	Level2SpeedUIOverlay \
 	AmoledDarkUIOverlay \
 	CharcoalDarkUIOverlay \
 	GrayDimDarkUIOverlay \
diff --git a/packages/overlays/Level1SpeedUIOverlay/Android.mk b/packages/overlays/Level1SpeedUIOverlay/Android.mk
new file mode 100644
index 00000000000..534f3f179ef
--- /dev/null
+++ b/packages/overlays/Level1SpeedUIOverlay/Android.mk
@@ -0,0 +1,30 @@
+#
+#  Copyright 2019, The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_RRO_THEME := Level1SpeedUI
+
+LOCAL_PRODUCT_MODULE := true
+
+LOCAL_SRC_FILES := $(call all-subdir-java-files)
+
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
+
+LOCAL_PACKAGE_NAME := Level1SpeedUIOverlay
+LOCAL_SDK_VERSION := current
+
+include $(BUILD_RRO_PACKAGE)
diff --git a/packages/overlays/Level1SpeedUIOverlay/AndroidManifest.xml b/packages/overlays/Level1SpeedUIOverlay/AndroidManifest.xml
new file mode 100644
index 00000000000..999f604cc52
--- /dev/null
+++ b/packages/overlays/Level1SpeedUIOverlay/AndroidManifest.xml
@@ -0,0 +1,24 @@
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.android.theme.speedui.level1"
+    android:versionCode="1"
+    android:versionName="1.0">
+    <overlay android:targetPackage="com.android.systemui" android:category="android.theme.customization.speedui" android:priority="1"/>
+    <application android:label="Level 1 (OLED)" android:hasCode="false"/>
+</manifest>
diff --git a/packages/overlays/Level1SpeedUIOverlay/res/values/config.xml b/packages/overlays/Level1SpeedUIOverlay/res/values/config.xml
new file mode 100644
index 00000000000..237813863c4
--- /dev/null
+++ b/packages/overlays/Level1SpeedUIOverlay/res/values/config.xml
@@ -0,0 +1,28 @@
+<!--
+/**
+ * Copyright (c) 2020, CAOS Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <!-- Whether or not a background should be drawn behind a notification. -->
+    <bool name="config_drawNotificationBackground">false</bool>
+
+    <!-- Whether or not child notifications that are part of a group will have shadows. -->
+    <bool name="config_enableShadowOnChildNotifications">false</bool>
+
+    <!-- Whether to clip notification contents with a rounded rectangle. Might be expensive on
+         certain GPU's and thus can be turned off with only minimal visual impact. -->
+    <bool name="config_notifications_round_rect_clipping">false</bool>
+</resources>
diff --git a/packages/overlays/Level2SpeedUIOverlay/Android.mk b/packages/overlays/Level2SpeedUIOverlay/Android.mk
new file mode 100644
index 00000000000..6c0367375ad
--- /dev/null
+++ b/packages/overlays/Level2SpeedUIOverlay/Android.mk
@@ -0,0 +1,30 @@
+#
+#  Copyright 2019, The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+LOCAL_PATH:= $(call my-dir)
+include $(CLEAR_VARS)
+
+LOCAL_RRO_THEME := Level2SpeedUI
+
+LOCAL_PRODUCT_MODULE := true
+
+LOCAL_SRC_FILES := $(call all-subdir-java-files)
+
+LOCAL_RESOURCE_DIR := $(LOCAL_PATH)/res
+
+LOCAL_PACKAGE_NAME := Level2SpeedUIOverlay
+LOCAL_SDK_VERSION := current
+
+include $(BUILD_RRO_PACKAGE)
diff --git a/packages/overlays/Level2SpeedUIOverlay/AndroidManifest.xml b/packages/overlays/Level2SpeedUIOverlay/AndroidManifest.xml
new file mode 100644
index 00000000000..dae7152ca4e
--- /dev/null
+++ b/packages/overlays/Level2SpeedUIOverlay/AndroidManifest.xml
@@ -0,0 +1,24 @@
+<!--
+/**
+ * Copyright (c) 2019, The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.android.theme.speedui.level2"
+    android:versionCode="1"
+    android:versionName="1.0">
+    <overlay android:targetPackage="com.android.systemui" android:category="android.theme.customization.speedui" android:priority="1"/>
+    <application android:label="Level 2 (IPS)" android:hasCode="false"/>
+</manifest>
diff --git a/packages/overlays/Level2SpeedUIOverlay/res/values/config.xml b/packages/overlays/Level2SpeedUIOverlay/res/values/config.xml
new file mode 100644
index 00000000000..e7b4473892b
--- /dev/null
+++ b/packages/overlays/Level2SpeedUIOverlay/res/values/config.xml
@@ -0,0 +1,34 @@
+<!--
+/**
+ * Copyright (c) 2020, CAOS Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<resources>
+    <!-- Whether or not a background should be drawn behind a notification. -->
+    <bool name="config_drawNotificationBackground">false</bool>
+
+    <!-- Whether or not child notifications that are part of a group will have shadows. -->
+    <bool name="config_enableShadowOnChildNotifications">false</bool>
+
+    <!-- Whether to clip notification contents with a rounded rectangle. Might be expensive on
+         certain GPU's and thus can be turned off with only minimal visual impact. -->
+    <bool name="config_notifications_round_rect_clipping">false</bool>
+
+    <!-- Whether to enable dimming navigation buttons when wallpaper is not visible, should be
+         enabled for OLED devices to reduce/prevent burn in on the navigation bar (because of the
+         black background and static button placements) and disabled for all other devices to
+         prevent wasting cpu cycles on the dimming animation -->
+    <bool name="config_navigation_bar_enable_auto_dim_no_visible_wallpaper">false</bool>
+</resources>
-- 
2.25.1

